pkg/framework/aries/api/vdr/create/option.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package create implements create did option
+//
+package create
+
+import (
+	"crypto"
+
+	docdid "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr/doc"
+)
+
+// Opts create did opts.
+type Opts struct {
+	PublicKeys             []doc.PublicKey
+	Services               []docdid.Service
+	GetEndpoints           func() ([]string, error)
+	RecoveryPublicKey      crypto.PublicKey
+	UpdatePublicKey        crypto.PublicKey
+	SigningKey             crypto.PrivateKey
+	SigningKeyID           string
+	MultiHashAlgorithm     uint
+	DefaultServiceType     string
+	DefaultServiceEndpoint string
+	EncryptionKey          *doc.PublicKey
+}
+
+// Option is a create DID option.
+type Option func(opts *Opts)
+
+// WithPublicKey add DID public key.
+func WithPublicKey(publicKey *doc.PublicKey) Option {
+	return func(opts *Opts) {
+		opts.PublicKeys = append(opts.PublicKeys, *publicKey)
+	}
+}
+
+// WithEndpoints get endpoints.
+func WithEndpoints(getEndpoints func() ([]string, error)) Option {
+	return func(opts *Opts) {
+		opts.GetEndpoints = getEndpoints
+	}
+}
+
+// WithService add service.
+func WithService(service *docdid.Service) Option {
+	return func(opts *Opts) {
+		opts.Services = append(opts.Services, *service)
+	}
+}
+
+// WithRecoveryPublicKey set recovery public key.
+func WithRecoveryPublicKey(recoveryPublicKey crypto.PublicKey) Option {
+	return func(opts *Opts) {
+		opts.RecoveryPublicKey = recoveryPublicKey
+	}
+}
+
+// WithUpdatePublicKey set update public key.
+func WithUpdatePublicKey(updatePublicKey crypto.PublicKey) Option {
+	return func(opts *Opts) {
+		opts.UpdatePublicKey = updatePublicKey
+	}
+}
+
+// WithMultiHashAlgorithm set multi hash algorithm for sidetree request.
+func WithMultiHashAlgorithm(multiHashAlgorithm uint) Option {
+	return func(opts *Opts) {
+		opts.MultiHashAlgorithm = multiHashAlgorithm
+	}
+}
+
+// WithDefaultServiceType service type of DID document to be created.
+func WithDefaultServiceType(serviceType string) Option {
+	return func(opts *Opts) {
+		opts.DefaultServiceType = serviceType
+	}
+}
+
+// WithDefaultServiceEndpoint service.
+func WithDefaultServiceEndpoint(endpoint string) Option {
```

Comment:
Can this option (plus type) be merged to with service endpoint?

Reply:
I will create PR

Reply:
done https://github.com/hyperledger/aries-framework-go/compare/8caf5cc4fe872997a0c88fb80f2be660a08eb20f..16d39ac0bb2479e1f043807c29019c6008c7042d